diff -Naur ./c++/drivers/lms/sicklms-1.0/SickLMS.cc ../../junk/sicktoolbox-1.0/c++/drivers/lms/sicklms-1.0/SickLMS.cc
--- ./c++/drivers/lms/sicklms-1.0/SickLMS.cc	2008-03-24 15:13:00.000000000 -0400
+++ ../../junk/sicktoolbox-1.0/c++/drivers/lms/sicklms-1.0/SickLMS.cc	2012-02-18 22:44:17.681665649 -0500
@@ -93,8 +93,9 @@
    * \brief Attempts to initialize the Sick LMS 2xx and then sets communication at
    *        at the given baud rate.
    * \param desired_baud_rate Desired session baud rate
+   * \param delay Delay to wait for SICK to power on
    */
-  void SickLMS::Initialize( const sick_lms_baud_t desired_baud_rate )
+  void SickLMS::Initialize( const sick_lms_baud_t desired_baud_rate, uint32_t delay )
     throw( SickConfigException, SickTimeoutException, SickIOException, SickThreadException ) {
 
     /* Buffer the desired baud rate in case we have to reset */
@@ -106,7 +107,7 @@
       
       /* Initialize the serial term for communication */
       std::cout << "\tAttempting to open device @ " << _sick_device_path << std::endl << std::flush;
-      _setupConnection();
+      _setupConnection(delay);
       std::cout << "\t\tDevice opened!" << std::endl << std::flush;
 
       /* Start/reset the buffer monitor */
@@ -2306,18 +2307,29 @@
     
   }
   
-  /**
+   /**
    * \brief Attempts to open a I/O stream using the device path given at object instantiation
    */
   void SickLMS::_setupConnection( ) throw ( SickIOException, SickThreadException ) {
+    SickLMS::_setupConnection( 0 );
+  }
+  
+  /**
+   * \brief Attempts to open a I/O stream using the device path given at object instantiation
+   * \param delay Delay to wait for SICK to power on.
+   */
+  void SickLMS::_setupConnection( uint32_t delay ) throw ( SickIOException, SickThreadException ) {
 
     try {
     
       /* Open the device */
-      if((_sick_fd = open(_sick_device_path.c_str(), O_RDWR | O_NOCTTY)) < 0) {
+      if((_sick_fd = open(_sick_device_path.c_str(), O_RDWR | O_NOCTTY | O_NDELAY)) < 0) {
 	throw SickIOException("SickLMS::_setupConnection: - Unable to open serial port");
       }
       
+      // Sleep to allow the SICK to power on for some applications
+      sleep(delay);
+      
       /* Backup the original term settings */
       if(tcgetattr(_sick_fd,&_old_term) < 0) {
 	throw SickIOException("SickLMS::_setupConnection: tcgetattr() failed!");
diff -Naur ./c++/drivers/lms/sicklms-1.0/SickLMS.hh ../../junk/sicktoolbox-1.0/c++/drivers/lms/sicklms-1.0/SickLMS.hh
--- ./c++/drivers/lms/sicklms-1.0/SickLMS.hh	2008-03-24 15:07:41.000000000 -0400
+++ ../../junk/sicktoolbox-1.0/c++/drivers/lms/sicklms-1.0/SickLMS.hh	2012-02-18 22:43:42.051687625 -0500
@@ -513,7 +513,7 @@
     ~SickLMS( );
     
     /** Initializes the Sick */
-    void Initialize( const sick_lms_baud_t desired_baud_rate )
+    void Initialize( const sick_lms_baud_t desired_baud_rate, uint32_t delay = 0 )
       throw( SickConfigException, SickTimeoutException, SickIOException, SickThreadException);
 
     /** Uninitializes the Sick */
@@ -750,6 +750,7 @@
 
     /** Opens the terminal for serial communication. */
     void _setupConnection( ) throw( SickIOException, SickThreadException );
+    void _setupConnection( uint32_t delay ) throw( SickIOException, SickThreadException );
 
     /** Closes the serial communication terminal. */
     void _teardownConnection( ) throw( SickIOException );
